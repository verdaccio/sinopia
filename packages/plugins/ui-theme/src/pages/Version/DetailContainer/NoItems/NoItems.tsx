import React from 'react';
import Text from 'verdaccio-ui/components/Text';

interface Props {
  text: string;
  className?: string;
}

const NoItems: React.FC<Props> = ({ className, text, ...props }) => (
  // eslint-disable-next-line verdaccio/jsx-spread
  <Text {...props} className={className} gutterBottom={true} variant="subtitle1">
    {text}
  </Text>
);

export default NoItems;
