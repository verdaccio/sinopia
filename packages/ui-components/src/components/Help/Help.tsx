import Button from '@mui/material/Button';
import CardActions from '@mui/material/CardActions';
import CardContent from '@mui/material/CardContent';
import Typography from '@mui/material/Typography';
import React, { Fragment } from 'react';
import { useTranslation } from 'react-i18next';

import { CopyClipboard, useConfig } from '../../';
import { Heading } from '../../';
import { CardStyled as Card } from './styles';

function renderHeadingClipboardSegments(title: string, text: string): React.ReactNode {
  return (
    <Fragment>
      <Typography variant={'body1'}>{title}</Typography>
      <CopyClipboard dataTestId="segments" text={text} />
    </Fragment>
  );
}

const Help: React.FC = () => {
  const { configOptions } = useConfig();
  const registryUrl = configOptions.base;
  const { t } = useTranslation();

  return (
    <Card data-testid="help-card" id="help-card">
      <CardContent>
        <Heading gutterBottom={true} id="help-card__title" variant="h5">
          {t('help.title')}
        </Heading>
        <Heading color="textSecondary" gutterBottom={true} sx={{ marginBottom: 1 }}>
          {t('help.sub-title')}
        </Heading>
        {renderHeadingClipboardSegments(
          t('help.first-step'),
          t('help.first-step-command-line', { registryUrl })
        )}
        {renderHeadingClipboardSegments(
          t('help.second-step'),
          t('help.second-step-command-line', { registryUrl })
        )}
        <Typography variant="body2">{t('help.third-step')}</Typography>
      </CardContent>
      <CardActions>
        <Button color="primary" href="https://verdaccio.org/docs/cli-registry" size="small">
          {t('button.learn-more')}
        </Button>
      </CardActions>
    </Card>
  );
};

export default Help;
