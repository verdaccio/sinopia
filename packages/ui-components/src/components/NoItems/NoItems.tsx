import Typography from '@mui/material/Typography';
import React from 'react';

interface Props {
  text: string;
  className?: string;
}

const NoItems: React.FC<Props> = ({ className, text, ...props }) => (
  // eslint-disable-next-line verdaccio/jsx-spread
  <Typography {...props} className={className} gutterBottom={true} variant="subtitle1">
    {text}
  </Typography>
);

export default NoItems;
