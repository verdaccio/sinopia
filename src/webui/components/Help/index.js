/**
 * @prettier
 * @flow
 */

import React, { Fragment } from 'react';
import type { Node } from 'react';
import CardActions from '@material-ui/core/CardActions/index';
import CardContent from '@material-ui/core/CardContent/index';
import Button from '@material-ui/core/Button/index';
import Typography from '@material-ui/core/Typography/index';

import CopyToClipBoard from '../CopyToClipBoard/index';
import { getRegistryURL } from '../../utils/url';
import { CardStyled as Card, HelpTitle } from './styles';

function renderHeadingClipboardSegments(title: string, text: string): Node {
  return (
    <Fragment>
      <Typography variant="body2">{title}</Typography>
      <CopyToClipBoard text={text} />
    </Fragment>
  );
}

const Help = (): Node => {
  const registryUrl = getRegistryURL();

  return (
    <Card id="help-card">
      <CardContent>
        <Typography component="h2" variant="headline" gutterBottom id="help-card__title">
          No Package Published Yet.
        </Typography>
        <HelpTitle color="textSecondary" gutterBottom>
          To publish your first package just:
        </HelpTitle>
        {renderHeadingClipboardSegments('1. Login', `$ npm adduser --registry ${registryUrl}`)}
        {renderHeadingClipboardSegments('2. Publish', `$ npm publish --registry ${registryUrl}`)}
        <Typography variant="body2">3. Refresh this page.</Typography>
      </CardContent>
      <CardActions>
        <Button href="https://verdaccio.org/docs/en/installation" target="_blank" size="small" color="primary">
          Learn More
        </Button>
      </CardActions>
    </Card>
  );
};

export default Help;
