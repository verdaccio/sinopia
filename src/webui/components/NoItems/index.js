/**
 * @prettier
 * @flow
 */

import React from 'react';
import Typography from '@material-ui/core/Typography/index';

import { IProps } from './types';

const NoItems = ({ text }: IProps) => (
  <Typography gutterBottom={true} variant={'subtitle1'}>
    {text}
  </Typography>
);

export default NoItems;
