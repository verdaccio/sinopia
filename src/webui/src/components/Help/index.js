
import React from 'react';
import SyntaxHighlighter, {registerLanguage} from 'react-syntax-highlighter/dist/light';
import sunburst from 'react-syntax-highlighter/src/styles/sunburst';
import js from 'react-syntax-highlighter/dist/languages/javascript';

import classes from './help.scss';

registerLanguage('javascript', js);

const Help = () => {
  // Don't add slash if it's not a sub directory
  const registryURL = `${location.origin}${location.pathname === '/' ? '' : location.pathname}`;

    return (
      <div className={classes.help}>
        <li className={classes.noPkg}>
          <h1 className={classes.noPkgTitle}>
            No Package Published Yet
          </h1>
          <div className={classes.noPkgIntro}>
            <div>
              To publish your first package just:
            </div>
            <br/>
            <strong>
              1. Login
            </strong>
            <SyntaxHighlighter language='javascript' style={sunburst}>
              {`npm adduser --registry  ${registryURL}`}
            </SyntaxHighlighter>
            <strong>2. Publish</strong>
            <SyntaxHighlighter language='javascript' style={sunburst}>
              {`npm publish --registry ${registryURL}`}
            </SyntaxHighlighter>
            <strong>3. Refresh this page!</strong>
          </div>
        </li>
      </div>
    );
};

export default Help;
