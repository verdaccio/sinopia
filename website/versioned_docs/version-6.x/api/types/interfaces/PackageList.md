---
id: 'PackageList'
title: 'Interface: PackageList'
sidebar_label: 'PackageList'
sidebar_position: 0
custom_edit_url: null
---

## Indexable

â–ª [key: `string`]: [`PackageAccess`](PackageAccess.md)
