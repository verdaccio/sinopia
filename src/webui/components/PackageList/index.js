import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import Divider from '@material-ui/core/Divider';

import Package from '../Package';
import Help from '../Help';
import { formatLicense } from '../../utils/package';

import classes from './packageList.scss';

export default class PackageList extends React.Component {
  static propTypes = {
    packages: PropTypes.array,
  };

  render() {
    return (
      <div className={"package-list-items"}>
        <div className={classes.pkgContainer}>
          {this.hasPackages() ? this.renderPackages(): <Help /> }
        </div>
      </div>
    );
  }

  hasPackages() {
    const {packages} = this.props;
    return packages.length > 0;
  }

  renderPackages = () => {
    return (
      <Fragment>
        {this.renderList()}
      </Fragment>
    );
  }

  renderList = () => {
    const { packages } = this.props;
    return (
      packages.map((pkg, i) => {
        const { name, version, description, time, keywords, dist, homepage, bugs } = pkg;
        const author = pkg.author;
        // TODO: move format license to API side.
        const license = formatLicense(pkg.license);
        return (
          <React.Fragment key={i}>
            {i !== 0 && <Divider></Divider>}
            <Package {...{ name, dist, version, author, description, license, time, keywords, homepage, bugs }} />
          </React.Fragment>
        );
      })
    );
  }
}
