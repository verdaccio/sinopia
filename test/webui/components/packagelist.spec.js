/**
 * PackageList component
 */

import React from 'react';
import { mount } from 'enzyme';
import PackageList from '../../../src/webui/src/components/PackageList';
import Help from '../../../src/webui/src/components/Help';
import NoItems from '../../../src/webui/src/components/NoItems';
import { BrowserRouter } from 'react-router-dom';

describe('<PackageList /> component', () => {
  it('should load the component with no packages', () => {
    const props = {
      packages: [],
      help: true
    };
    const wrapper = mount(
      <PackageList packages={props.packages} help={props.help} />
    );

    const instance = wrapper.instance();
    expect(instance.isTherePackages()).toBeFalsy();
    expect(instance.renderHelp()).toBeTruthy();
    expect(instance.renderOptions()).toEqual(<Help />);
    expect(instance.renderTitle()).toBeUndefined();
  });

  it('should load the component with packages', () => {
    const props = {
      packages: [
        {
          name: 'verdaccio',
          version: '1.0.0',
          description: 'Private NPM repository',
          author: { name: 'Sam' }
        },
        {
          name: 'abc',
          version: '1.0.1',
          description: 'abc description',
          author: { name: 'Rose' }
        },
        {
          name: 'xyz',
          version: '1.1.0',
          description: 'xyz description',
          author: { name: 'Martin' }
        }
      ],
      help: false
    };
    const wrapper = mount(
      <BrowserRouter>
        <PackageList packages={props.packages} help={props.help} />
      </BrowserRouter>
    );

    const instance = wrapper.find(PackageList).instance();

    expect(instance.isTherePackages()).toBeTruthy();
    expect(instance.renderHelp()).toBeUndefined();
    expect(instance.renderTitle().props.children).toEqual('Available Packages');
    expect(instance.renderNoItems()).toEqual(
      <NoItems text="No items were found with that query" />
    );
    expect(instance.renderOptions()).toEqual(
      <NoItems text="No items were found with that query" />
    );
    // package count
    expect(wrapper.find('Package')).toHaveLength(3);
    // match snapshot
    expect(wrapper.html()).toMatchSnapshot();
  });
});
