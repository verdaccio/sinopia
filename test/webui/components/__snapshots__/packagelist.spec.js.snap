// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PackageList /> component should load the component with packages 1`] = `"<div><div class=\\"pkgContainer\\"><h1 class=\\"listTitle\\">Available Packages</h1><li><a class=\\"package\\" href=\\"detail/verdaccio\\"><h1>verdaccio<span class=\\"el-tag el-tag--gray\\">v1.0.0</span></h1><span role=\\"author\\" class=\\"author\\">By: Sam</span><p>Private NPM repository</p></a></li><li><a class=\\"package\\" href=\\"detail/abc\\"><h1>abc<span class=\\"el-tag el-tag--gray\\">v1.0.1</span></h1><span role=\\"author\\" class=\\"author\\">By: Rose</span><p>abc description</p></a></li><li><a class=\\"package\\" href=\\"detail/xyz\\"><h1>xyz<span class=\\"el-tag el-tag--gray\\">v1.1.0</span></h1><span role=\\"author\\" class=\\"author\\">By: Martin</span><p>xyz description</p></a></li></div></div>"`;
